pkg/kcp/informers.go

```diff
@@ -0,0 +1,105 @@
+package kcp
+
+import (
+	"reflect"
+	"time"
+	"unsafe"
+
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type clusterAwareSharedIndexInformer struct {
+	delegate cache.SharedIndexInformer
+	keyFunc  cache.KeyFunc
+}
+
+func NewClusterAwareSharedIndexInformerFunc(keyFunc cache.KeyFunc) client.NewInformerFunc {
+	//TODO(kcp) could hardcode kcp indexers here or leave them plumbed through so they are passed when this function is called
+	return func(lw cache.ListerWatcher, exampleObject k8sruntime.Object, defaultEventHandlerResyncPeriod time.Duration, indexers cache.Indexers) cache.SharedIndexInformer {
+		return &clusterAwareSharedIndexInformer{
+			delegate: cache.NewSharedIndexInformer(lw, exampleObject, defaultEventHandlerResyncPeriod, indexers),
+			keyFunc:  keyFunc,
+		}
+	}
+}
+
+func (i *clusterAwareSharedIndexInformer) AddEventHandler(handler cache.ResourceEventHandler) {
+	i.delegate.AddEventHandler(handler)
+}
+
+func (i *clusterAwareSharedIndexInformer) AddEventHandlerWithResyncPeriod(handler cache.ResourceEventHandler, resyncPeriod time.Duration) {
+	i.delegate.AddEventHandlerWithResyncPeriod(handler, resyncPeriod)
+}
+
+func (i *clusterAwareSharedIndexInformer) GetStore() cache.Store {
+	return i.delegate.GetStore()
+}
+
+func (i *clusterAwareSharedIndexInformer) GetController() cache.Controller {
+	return i.delegate.GetController()
+}
+
+func (i *clusterAwareSharedIndexInformer) Run(stopCh <-chan struct{}) {
+	go i.delegate.Run(stopCh)
+	// TODO(kcp) this is fragile
+	time.Sleep(10000)
```

Comment:
need to do something smarter here, issue is that the controller isn't initialized yet so we get a bunch of zero value errors. Trying to find a good way to wait for the fields we're setting to be initialized.